### PR TITLE
feat(common): add loaders for Cloudinary and ImageKit providers

### PR DESCRIPTION
**Pull request by khempenius**
## PR Type
What kind of change does this PR introduce?
- [x] Feature

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/common/src/directives/ng_optimized_image/image_loaders/cloudinary_loader.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      if (config.width) {
+        params += `,w_${config.width}`;
+      }
+      const url = `${path}/image/upload/${params}/${normalizeSrc(config.src)}`;
+      return url;
+    }
+  };
+}
+
+function assertValidPath(path: unknown) {
```
Comment (pkozlowski-opensource) [inline packages/common/src/directives/ng_optimized_image/image_loaders/cloudinary_loader.ts:47]: The code from here down to the end of this file is duplicated between loaders (similar code already [exists in the repo already](https://github.com/angular/angular/blob/7173711d16668003a41f095c615e39ae125f760a/packages/common/src/directives/ng_optimized_image/image_loaders/imgix_loader.ts#L39-L67)).

We should extract those utilities into something like loader_utils.ts or similar.

File: packages/common/src/directives/ng_optimized_image/image_loaders/imagekit_loader.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      if (config.width) {
+        params += `,w-${config.width?.toString()}`;
+      };
+      const url = `${path}/${params}/${normalizeSrc(config.src)}`;
+      return url;
+    }
+  };
+}
+
+function assertValidPath(path: unknown) {
```
Comment (pkozlowski-opensource) [inline packages/common/src/directives/ng_optimized_image/image_loaders/imagekit_loader.ts:46]: The code from here down to the end of this file is duplicated between loaders (similar code already [exists in the repo already](https://github.com/angular/angular/blob/7173711d16668003a41f095c615e39ae125f760a/packages/common/src/directives/ng_optimized_image/image_loaders/imgix_loader.ts#L39-L67)).

We should extract those utilities into something like loader_utils.ts or similar.

**Review by pkozlowski-opensource — CHANGES_REQUESTED** [review]
The change LGTM overall but we start to have code duplication between various loaders and I would love to see the common logic extracted before this PR gets merged.

File: packages/common/src/directives/ng_optimized_image/image_loaders/cloudinary_loader.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    throwInvalidPathError(path);
+  }
+  path = normalizePath(path);
+
+  return {
+    provide: IMAGE_LOADER,
+    useValue: (config: ImageLoaderConfig) => {
+      // Example of a Cloudinary image URL:
+      // https://res.cloudinary.com/mysite/image/upload/c_scale,f_auto,q_auto,w_600/marketing/tile-topics-m.png
+      let params = `f_auto,q_auto`;
```
Comment (kara) [inline packages/common/src/directives/ng_optimized_image/image_loaders/cloudinary_loader.ts:38]: Can we add a brief comment for what these options mean and why we're adding them?

File: packages/common/src/directives/ng_optimized_image/image_loaders/imagekit_loader.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    throwInvalidPathError(path);
+  }
+  path = normalizePath(path);
+
+  return {
+    provide: IMAGE_LOADER,
+    useValue: (config: ImageLoaderConfig) => {
+      // Example of an ImageKit image URL:
+      // https://ik.imagekit.io/demo/tr:w-300,h-300/medium_cafe_B1iTdD0C.jpg
+      let params = `tr:q-auto`;
```
Comment (kara) [inline packages/common/src/directives/ng_optimized_image/image_loaders/imagekit_loader.ts:37]: Comment for what this param does?

**Review by kara — APPROVED** [review]
LGTM, minor nits below

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

**Pull request merged**